src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java

```diff
@@ -2,33 +2,35 @@
 
 import java.net.MalformedURLException;
 
+import com.browserstack.examples.config.DriverType;
+import com.browserstack.examples.config.WebDriverConfiguration;
+import lombok.SneakyThrows;
 import org.junit.rules.TestWatcher;
 import org.junit.runner.Description;
 import org.openqa.selenium.JavascriptExecutor;
 import org.openqa.selenium.WebDriver;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
 import com.browserstack.examples.config.Platform;
 import com.browserstack.examples.config.WebDriverFactory;
 
+
 /**
  * Created with IntelliJ IDEA.
  *
  * @author Anirudha Khanna
  */
 public class WebDriverProviderRule extends TestWatcher {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(WebDriverProviderRule.class);
-
     private static final String TEST_STATUS_SCRIPT = "browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"%s\", \"reason\": \"%s\"}}";
-
     private String methodName;
     private WebDriver driver;
     private final WebDriverFactory webDriverFactory;
+    private WebDriverConfiguration webDriverConfiguration;
```

Comment:
Do not add this. Add the required methods to the WebDriverFactory itself.

Reply:
Done , i have removed   private WebDriverConfiguration webDriverConfiguration and added method in WebDriverFactory class